### ISSUE TITLE
No qualifier concept ids defined in the ValueSet Mapping document 

### ISSUE THREAD

**Issue by stephanieshong** [National-COVID-Cohort-Collaborative/Data-Ingestion-and-Harmonization#55]
Please validate if this is correct: 
case when lower(trim(result_qual)) = 'positive' then 45884084
            when lower(trim(result_qual)) = 'negative' then 45878583
            when lower(trim(result_qual)) = 'normal' then 45884153
            when lower(trim(result_qual)) = 'abnormal' then 45878745
            when lower(trim(result_qual)) = 'low' then 45881666
            when lower(trim(result_qual)) = 'high' then 45876384
            when lower(trim(result_qual)) = 'borderline' then 45880922
            when lower(trim(result_qual)) = 'elevatied' then 3445772 --ssh add issue number 
            when lower(trim(result_qual)) = 'un' then 45880480
            else 45877393
            end as QUALIFIER_CONCEPT_ID --null/un/elevated/boderline/ot/low/high/normal/negative/positive/abnormal

**Comment by kmkostka** [comment]
These are fine. 

Elevated using Concept ID: 3445772 uis a non-standard vocabulary (Nebraska Lexicon). That source concept maps to the Concept ID: 4328749 (https://athena.ohdsi.org/search-terms/terms/4328749) where High replaces Elevated (https://athena.ohdsi.org/search-terms/terms/40287622) as of 01/01/2013.

**Comment by stephanieshong** [comment]
Thank you. @kmkostka  Does this mean other values are ok?

**Comment by stephanieshong** [comment]
LAB_RESULT_CM | RESULT_QUAL | ABNORMAL | Abnormal | See notes | Gap
-- | -- | -- | -- | -- | --
LAB_RESULT_CM | RESULT_QUAL | BORDERLINE | Borderline | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | ELEVATED | Elevated | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | HIGH | High | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | LOW | Low | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | NEGATIVE | Negative | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | NI | No  Information | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | NORMAL | Normal | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | OT | Other | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | POSITIVE | Positive | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | UN | Unknown | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | UNDETECTABLE | Undetectable | See notes | Gap
LAB_RESULT_CM | RESULT_QUAL | UNDETERMINED | Undetermined | See notes | Gap

What concept id should we use for undetectable and undetermined?

**Comment by cyaghmour** [comment]
Hi,

In her review, Kristen suggested using concept_id zero.

Thanks,
Charles


________________________________

**Comment by stephanieshong** [comment]
@cyaghmour - could you validate the following numbers?

when lower(trim(result_qual)) = 'un' then 45880480
else 45877393<--- default value

**Comment by stephanieshong** [comment]
https://github.com/National-COVID-Cohort-Collaborative/Data-Ingestion-and-Harmonization/issues/55

Could you double confirm concept Id that I populaed on this page?

Stephanie

**Comment by stephanieshong** [comment]
Hi Charles,
In your valueSet mapping document you usually provide all the following column I have listed below.
Could you do the same thing for the following list of qualitative codes:
I cannot populate the table with missing information.


CDM_TBL

CDM_TBL_COLUMN_NAME

SRC_CODE

SRC_CD_DESCRIPTION

TARGET_CONCEPT_ID

TARGET_CONCEPT_NAME

TARGET_DOMAIN_ID

TARGET_VOCABULARY_ID

TARGET_CONCEPT_CLASS_ID

TARGET_STANDARD_CONCEPT

TARGET_CONCEPT_CODE



LAB_RESULT_CM

RESULT_QUAL

ABNORMAL

Abnormal

45878745

LAB_RESULT_CM

RESULT_QUAL

BORDERLINE

Borderline

45880922

LAB_RESULT_CM

RESULT_QUAL

ELEVATED

Elevated

4328749

LAB_RESULT_CM

RESULT_QUAL

HIGH

High

45876384

LAB_RESULT_CM

RESULT_QUAL

LOW

Low

45881666

LAB_RESULT_CM

RESULT_QUAL

NEGATIVE

Negative

45878583

LAB_RESULT_CM

RESULT_QUAL

NI

No  Information

46237210

LAB_RESULT_CM

RESULT_QUAL

NORMAL

Normal

45884153

LAB_RESULT_CM

RESULT_QUAL

OT

Other

46237210

LAB_RESULT_CM

RESULT_QUAL

POSITIVE

Positive

45884084

LAB_RESULT_CM

RESULT_QUAL

UN

Unknown

45877986

LAB_RESULT_CM

RESULT_QUAL

UNDETECTABLE

Undetectable

0

LAB_RESULT_CM

RESULT_QUAL

UNDETERMINED

Undetermined

0



Stephanie

**Comment by cyaghmour** [comment]
Hi @stephanieshong ,

I am no longer the custodian of the mapping spreasheet so I can't update it. Please check with Davera. However, I have attached a stand alone file with the information you requested.

Thanks
Charles
[lab_results_cm_results_qual-only.xlsx](https://github.com/National-COVID-Cohort-Collaborative/Data-Ingestion-and-Harmonization/files/4914544/lab_results_cm_results_qual-only.xlsx)

**Comment by stephanieshong** [comment]
@DaveraGabriel @cyaghmour @kmkostka @cukarthik 
RESULT_QUAL | UNDETECTABLE | Undetectable | 0
-- | -- | -- | --


For Undetectable could we use unknown instead of 0?

**Comment by stephanieshong** [comment]
@DaveraGabriel<https://github.com/DaveraGabriel> @cyaghmour<https://github.com/cyaghmour> @kmkostka<https://github.com/kmkostka> @cukarthik<https://github.com/cukarthik>
RESULT_QUAL

UNDETECTABLE

Undetectable

0

For Undetectable could we use unknown ( 45877393 unknown concept _id ) instead of 0 for concept id?
0 seems that we did not map to anything.
Stephanie

**Comment by tannerzhang** [comment]
'pos' 45884084
'neg' 45878583
'presumptive positive' 45884084
'presumptive negative'45878583
'detected'  45884084
'not detected'  45878583
'inconclusive' 45877990

The mapping process is not case-sensitive and the spaces in the qualitative result fields are trimmed.

**Comment by kmkostka** [comment]
@stephanieshong The concept you picked is related to a LOINC code that is supposed to be used as [an answer for Race values](https://athena.ohdsi.org/search-terms/terms/45877393).

Which is to say: you could map it to this value but there’s a chance most end users would not know to use this VALUE_AS_A_CONCEPT when they were specifying a query. An end user would be looking to pull LOINC codes that are expected to be results for a lab test.

**Comment by stephanieshong** [comment]
@kmkostka - Can you be more specific? We are not using,  45877393, anywhere.

**Comment by kmkostka** [comment]
@stephanieshong You tagged me above and asked: “For Undetectable could we use unknown ( 45877393 unknown concept _id ) instead of 0 for concept id?”

My response is: this code is probably not appropriate.

**Comment by stephanieshong** [comment]
'un' map to 0 
 'unknown' map to 0
undetectable should be map to negative  45878583 ?
@kmkostka  - is this correct?

**Comment by kmkostka** [comment]
@stephanieshong Yes for ‘un’ and ‘unknown’. Not sure Undetected is clinically synonymous with Negative. At least in HIV research, it definitely isn’t. I’m not sure what the prevailing clinical wisdom is for COVID. @cgchute or @cukarthik, thoughts?

**Comment by DaveraGabriel** [comment]
My apologies for not uploading this work prior to just now.  Last week, I took a stab at mapping these nulls that were produced as a result of running the DQD for one site.  Had planned to discus in our mapping validation meeting scheduled for tomorrow.  The urgency of the N3C soft launch has preempted this delay.  
[fkDomain Check Failure Concept Codes_DG_Comments6JUL2020.xlsx]

**Comment by stephanieshong** [comment]
Other, No Information, and Unknowns are now mapped to 0 . (OT, NI, and UN are mapped to 0)
